blocks/api/post.pegjs

```diff
@@ -296,12 +352,19 @@ Block_Name_Part
   = $( [a-z][a-z0-9_-]* )
 
 Block_Attributes
-  = attrs:$("{" (!("}" WS+ """/"? "-->") .)* "}")
+  = attrs:$("{" (!("}") .)* "}")
```

Comment:
This change was necessary. Without this, the parser was not parsing the blocks with attributes and version properly. 

I'm aware this change is not good because it means the "JSON" can't contain "}" at all. Would love some help here @aduth @dmsnell

Reply:
I'd strongly discourage the changes as they are because I think it overcomplicates the parser _and_ it breaks JSON.

The nice thing about the system we were using is that it fits within normal HTML comments without breaking and only imposes the additional constraint that `--` be escaped, which would be necessary regardless of JSON since we're already inside an HTML comment.

immediately I can imagine a few alternatives here:
 - put the version string before the attributes. this requires few changes and would be quite easy in the grammar spec. the version being prefixed would immediately fail when not present, so we can just add it in before the attributes as an optional value
 - store the version inside the attributes - no changes to the parser are even required.

Reply:
It's worth noting too that the existing code is a massive optimization. We could support a full JSON parse here and not worry about eating all the input until the block header ends, but then we'd have to include a full JSON parser in this grammar. As-is, we do a fast scan to the end then feed in the text to the native JSON parser built in the browser or in PHP which will be magnitudes faster than if we hand-built one.

Reply:
I tried something like

```
Block_Header
  = "<!--" WS+ "wp:" blockName:Block_Name WS+ version:(v:Block_Version WS+ {
    /** <?php return $v; ?> **/
    return $v;
  })? attrs:(a:Block_Attributes WS+ {
    /** <?php return $a; ?> **/
    return a;
  })? "-->"
  {
    /** <?php
    return array(
      'blockName' => $blockName,
      'attrs'     => $attrs,
      'version'  => $version ? $version : 1,
    );
    ?> **/

    return {
      blockName: blockName,
      attrs: attrs,
      version: version
    };
  }
```

Similar to what we had before, but it's not working and I'm not sure what I am doing wrong.

Reply:
Oh! I was able to revert the JSON parsing change by inverting the version and the attributes.

Reply:
What is problematic about storing the version in the attributes?

Reply:
Nothing aside the "version" becomes a special attribute and we should guard against its usage in regular attributes.

What's problematic without storing it outside the attributes, since it's not an attribute semantically speaking?

Just to be clear, I'm not against this change.

Reply:
It’s a special attribute for sure, though with a stunningly obvious meaning and one we could easily communicate about and guard against (for example, but stripping it before blocks get their attributes).

I’d say the _cost_ is adding complexity to the parser where we already have a system for storing attributes. I’m not sure I’d argue that the version _isn’t_ an attribute.

Of course it’s fine if this is one of those intrinsic things every block has and is duly different than attributes to store it elsewhere, though I’m guessing it only feels different because we already combined separate attributes in the HTML style into the JSON structure in order to be more terse and eliminate escaping-confusion.

How would this be different if attributes had the same kind of key-value pairs in the block header?